src/udfs/ssd_object_detector.py

```diff
@@ -0,0 +1,375 @@
+import itertools
+import numpy as np
+import pandas as pd
+
+from typing import List
+from math import sqrt
+
+import torch
+import torch.nn.functional as F
+from torch import Tensor
+from torchvision.transforms import Compose, transforms
+
+from src.models.catalog.frame_info import FrameInfo
+from src.models.catalog.properties import ColorSpace
+from src.models.inference.outcome import Outcome
+from src.udfs.pytorch_abstract_udf import PytorchAbstractUDF
+
+
+class SSDObjectDetector(PytorchAbstractUDF):
+    @property
+    def name(self) -> str:
+        return "ssd"
+
+    def __init__(self, threshold=0.5):
+        super().__init__()
+        self.threshold = threshold
+
+        # load ssd from pytorch hub api and default to cpu
+        self.model = torch.hub.load(
+            "NVIDIA/DeepLearningExamples:torchhub",
+            "nvidia_ssd",
+            pretrained=False)
+        model_state_dict = torch.hub.load_state_dict_from_url(
+            "https://api.ngc.nvidia.com/v2/models/nvidia/ssdpyt_fp32/versions/1/files/nvidia_ssdpyt_fp32_20190225.pt",
+            map_location=torch.device("cpu"))["model"]
+        self.model.load_state_dict(model_state_dict)
```

Comment:
Will torch cache the model and downloaded weights? We don't want these downloads to happen at each call.

Reply:
Yes, it will cache.